src/postprocess/refine.jl

```diff
@@ -172,14 +172,13 @@ The refined energies can be obtained by E + dE.
 """
 function refine_energies(refinement::RefinementResult{T}) where {T}
     term_names = [string(nameof(typeof(term))) for term in refinement.basis.model.term_types]
-    result = DiffResults.DiffResult(zeros(T, length(term_names)),
-                                    zeros(T, length(term_names)))
+
     f(ε) = energy(refinement.basis,
                   refinement.ψ + ε.*refinement.δψ,
                   refinement.occupation;
                   ρ=refinement.ρ + ε.*refinement.δρ).energies.values
-    result = ForwardDiff.derivative!(result, f, zero(T))
-    (; E=Energies(term_names, result.value), dE=Energies(term_names, result.derivs[1]))
+    result = value_and_derivative(f, AutoForwardDiff(), zero(T))
+    (; E=Energies(term_names, result[1]), dE=Energies(term_names, result[2]))
```

Comment:
I don't like the `[1]` and `[2]` thing here. Can you not unpack the tuple (maybe it's even a named tuple) ?

Reply:
Better now?